pkg/aws/apis/validation/validation.go

```diff
@@ -191,6 +193,29 @@ func validateNetworkInterfaces(networkInterfaces []awsapi.AWSNetworkInterfaceSpe
 	return allErrs
 }
 
+func validateInstanceMetadata(metadata *awsapi.InstanceMetadataOptions, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if metadata == nil {
+		return allErrs
+	}
+
+	if metadata.HttpPutResponseHopLimit != nil {
+		if *metadata.HttpPutResponseHopLimit < 0 || *metadata.HttpPutResponseHopLimit > 64 {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("httpPutResponseHopLimit"), *metadata.HttpPutResponseHopLimit, "Only values between 0 and 64 are accepted"))
+		}
+
+		if metadata.HttpEndpoint != nil && *metadata.HttpEndpoint != "disabled" {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("httpEndpoint"), *metadata.HttpEndpoint, "Only 'disabled' is valid value"))
+		}
+
+		if metadata.HttpTokens != nil && *metadata.HttpTokens != "required" {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("httpTokens"), *metadata.HttpTokens, "Only 'required' is valid value"))
```

Comment:
`optional` is also valid. Yes its default right now, so anyone would only set `required` but later `requried` could become default value.